sorc/ncep_post.fd/MDLFLD.f

```diff
@@ -3980,33 +3775,34 @@ SUBROUTINE MDLFLD
           DO I=1,IM
 !Initialed as 'undetected'.  Nov. 17, 2014, B. ZHOU:
 !changed from SPVAL to -5000. to distinguish missing grids and undetected 
+            loop201: do
 !           GRID1(I,J) = SPVAL      	      
             GRID1(I,J) = -5000.  !undetected initially         
             IF(IMP_PHYSICS == 8.)then ! If Thompson MP
               DO L=1,NINT(LMH(I,J))
                 IF(REF_10CM(I,J,L) > 18.3) then
                   GRID1(I,J) = ZMID(I,J,L)
-                  go to 201
+!                 go to 201
+                  exit loop201
                 ENDIF
               ENDDO
             ELSE ! if other MP than Thompson
               DO L=1,NINT(LMH(I,J))
                 IF(DBZ(I,J,L) > 18.3) then
                   GRID1(I,J) = ZMID(I,J,L)
-                  go to 201
+!                 go to 201
+                  exit loop201
                 END IF
               ENDDO
             END IF
+            exit loop201
+            enddo loop201
```

Comment:
Would it work with the changes:
-----------------------------------------------
loop201:  IF(IMP_PHYSICS == 8.)then ! If Thompson MP
              DO L=1,NINT(LMH(I,J))
                IF(REF_10CM(I,J,L) > 18.3) then
                  GRID1(I,J) = ZMID(I,J,L)
                  exit loop201
                ENDIF
              ENDDO
            ELSE  loop201 ! if other MP than Thompson
              DO L=1,NINT(LMH(I,J))
                IF(DBZ(I,J,L) > 18.3) then
                  GRID1(I,J) = ZMID(I,J,L)
                  exit loop201
                END IF loop201
              ENDDO
            END IF

Reply:
Wen,

The above changes will cause compiling errors. The structure should be the below:

loop201: IF(IMP_PHYSICS == 8.)then ! If Thompson MP
DO L=1,NINT(LMH(I,J))
IF(REF_10CM(I,J,L) > 18.3) then
GRID1(I,J) = ZMID(I,J,L)
exit loop201
ENDIF
ENDDO
ELSE loop201 ! if other MP than Thompson
DO L=1,NINT(LMH(I,J))
IF(DBZ(I,J,L) > 18.3) then
GRID1(I,J) = ZMID(I,J,L)
exit loop201
END IF 
ENDDO
END IF loop201

There should be no compile error but will affect the "do loop" structure if they are compared with my version. I will talk to you more on meeting.

Reply:
Would this be straight forward:
            GRID1(I,J) = -5000.  !undetected initially
            IF(IMP_PHYSICS == 8.)then ! If Thompson MP
              DO L=1,NINT(LMH(I,J))
                IF(REF_10CM(I,J,L) > 18.3) then
                  GRID1(I,J) = ZMID(I,J,L)
!                 go to 201
                  exit
                ENDIF
              ENDDO
            ELSE ! if other MP than Thompson
              DO L=1,NINT(LMH(I,J))
                IF(DBZ(I,J,L) > 18.3) then
                  GRID1(I,J) = ZMID(I,J,L)
!                 go to 201
                  exit
                END IF
              ENDDO
            END IF
-------------------------------------

Reply:
I just feel a lots of do loops which have no loop control, specially in nested loops would easily make people lost during debugging.

Reply:
Yes, looks good to me.